examples/imagenet/alexnet.cc

```diff
@@ -0,0 +1,403 @@
+  net.Add(new Dense(), GenDenseConf("ip7", 4096, 0.005, 1, 1.0));
+  net.Add(new CudnnActivation(), GenReLUConf("relu7"));
+  net.Add(new CudnnDropout(), GenDropoutConf("drop7", 0.5));
+  net.Add(new Dense(), GenDenseConf("ip8", 1000, 0.01, 1));
+
+  return net;
+}
+
+void TrainOneEpoch(FeedForwardNet &net, ILSVRC &data,
+                                      std::shared_ptr<Device> device, int epoch,
+                                      string bin_folder,
+                                      size_t num_train_images,
+                                      size_t train_file_size,
+                                      size_t read_size,
+                                      float lr,
+                                      Channel *train_ch) {
+  size_t num_train_files = num_train_images / train_file_size +
+                           (num_train_images % train_file_size ? 1 : 0);
+  string mean_path = bin_folder + "/mean.bin";
+  float loss = 0.0f, metric = 0.0f;
+  float load_time = 0.0f, train_time = 0.0f;
+  size_t b = 0;
+  size_t n_read;
+  Timer timer, ttr;
+  Tensor prefetch_x, prefetch_y;
+  //prefetch_x.ToDevice(device);
+  //prefetch_y.ToDevice(device);
+  string binfile = bin_folder + "/train1.bin";
+  timer.Tick();
+  data.LoadData(kTrain, binfile, read_size, &prefetch_x, &prefetch_y, &n_read);
+  load_time += timer.Elapsed();
+  CHECK_EQ(n_read, read_size);
+  Tensor train_x(prefetch_x.shape(), device);
+  Tensor train_y(prefetch_y.shape(), device, kInt);
+  std::thread th;
+  //LOG(INFO) << "Total num of training files: " << num_train_files;
+  for (size_t fno = 1; fno <= num_train_files; fno++) {
+    binfile = bin_folder + "/train" + std::to_string(fno) + ".bin";
+    //LOG(INFO) << "Load data from " << binfile;
+    while (true) {
+      if (th.joinable()) {
+        th.join();
+        load_time += timer.Elapsed();
+        //LOG(INFO) << "num of samples: " << n_read;
+        if (n_read < read_size) {
+          if (n_read > 0) {
+            LOG(WARNING) << "Pls set batchsize to make num_total_samples "
+              << "% batchsize == 0. Otherwise, the last " << n_read
+              << " samples would not be used";
+          }
+          break;
+        }
+      }
+      if (n_read == read_size) {
+        train_x.CopyData(prefetch_x);
+        train_y.CopyData(prefetch_y);
+      }
+      //LOG(INFO) << "train_x.L1(): " << train_x.L1();
+      timer.Tick();
+      th = data.AsyncLoadData(kTrain, binfile, read_size, &prefetch_x, &prefetch_y, &n_read);
+      if (n_read < read_size) continue;
+      CHECK_EQ(train_x.shape(0), train_y.shape(0));
+ //     train_x.ToDevice(device);
+//      train_y.ToDevice(device);
+      ttr.Tick();
+      auto ret = net.TrainOnBatch(epoch, train_x, train_y);
+      train_time += ttr.Elapsed();
+      loss += ret.first;
+      metric += ret.second;
+      b++;
+      //LOG(INFO) << "batch " << b << ", loss: " << ret.first;
+    }
+    if (fno % 20 == 0) {
```

Comment:
pls use num of iterations as the print frequency. 
the fno would change if the train_file_size changes.